src/Dropdown.js

```diff
@@ -112,10 +112,11 @@ const Dropdown = React.forwardRef((uncontrolledProps, ref) => {
   );
 
   const handleSelect = useEventCallback((key, event) => {
-    if (onSelectCtx) onSelectCtx(key, event);
+    if (onSelectCtx && onSelectCtx.isDropdownSelCtx) onSelectCtx(key, event);
     if (onSelect) onSelect(key, event);
     handleToggle(false, event, 'select');
   });
+  handleSelect.isDropdownSelCtx = true;
```

Comment:
how does mutating the result of `useEventCallback` do anything to the value of `useContext(SelectableContext)` above?

Reply:
in other words you're just hard-disabling this functionality, which is wrong

Reply:
The idea is to modify the the context it provides to child components via
```JavaScript
<SelectableContext.Provider value={handleSelect}>
```
This way onSelectCtx will only fire if the parent is a Dropdown, closing that parent. I checked and this fix does break NavDropdown, so I will try another approach. I also found that selecting a Nav.Item will cause an Accordion to collapse as well. So perhaps Accordion would be a better place to fix this issue.

I still think a unit test for nested Dropdown behavior is good to have as this behavior is provided by bootstrap via the dropdown-submenu class.

Reply:
you're mutating the return value of a `useEventCallback` hook. and you shouldn't be mutating used context anyway